spec/unit/x509/cert_provider_spec.rb

```diff
@@ -0,0 +1,305 @@
+require 'spec_helper'
+require 'puppet/x509'
+
+describe Puppet::X509::CertProvider do
+  include PuppetSpec::Files
+
+  def pem(name)
+    File.read(my_fixture(name))
+  end
+
+  def cert(name)
+    OpenSSL::X509::Certificate.new(pem(name))
+  end
+
+  def crl(name)
+    OpenSSL::X509::CRL.new(pem(name))
+  end
+
+  def key(name)
+    OpenSSL::PKey::RSA.new(pem(name))
+  end
+
+  def request(name)
+    OpenSSL::X509::Request.new(pem(name))
+  end
+
+  def create_provider(options)
+    described_class.new(options)
+  end
+
+  context 'when loading' do
+    context 'cacerts' do
+      it 'returns nil if it does not exist' do
+        provider = create_provider(capath: '/does/not/exist')
+
+        expect(provider.load_cacerts).to be_nil
+      end
+
+      it 'returns an array of certificates' do
+        certs = create_provider(capath: my_fixture('ca.pem')).load_cacerts
+        expect(certs.count).to eq(1)
```

Comment:
would it be helpful to check that it's an array object rather than checking the count? Or do you think this is sufficient to verify we've been given an array?

Reply:
I updated this to use `contain_exactly` and `an_object_having_attributes`.